# test-datasets: `radseq`

This branch contains test data for [nf-core/radseq](blah) pipeline

## Content of this repository

`testdata/PRJNA386062`: ezRAD sequences from...

### For Reference Based Approach Or Denovo

#### Simulated Horseshoe Crab Data

`testdata/Lpolyphemus/ddrad_sbfi_mluci_chr26/hsc_Chr26.fasta` chromosome 26 of Atlantic Horseshoe Crab

`testdata/Lpolyphemus/ddrad_sbfi_mluci_chr26/msp_[0..9].{1,2}.fq.gz` simulated ddRAD reads

`testdata/Lpolyphemus/ddrad_sbfi_mluci_chr26/popmap.txt` id and corresponding population

`samplesheet/samplesheet_test_lpolyphemus_ddrad_sbfi_mluci_chr26.csv` input sample sheet for this dataset

#### Arctic Grayling 

`testdata/ddRAD/CM014990.1.fna`: Chromosome 1 

`testdata/ddRAD/Golden3B05.{1,2}.fq.gz`: individual 1 ddRAD seq. w/out umi barcodes

`testdata/ddRAD/Golden3B08.{1,2}.fq.gz`: individual 2 ddRAD seq. w/out umi barcodes

`testdata/ddRAD/Golden3B12.{1,2}.fq.gz`: individual 3 ddRAD seq. w/out umi barcodes

`testdata/ddRAD/Golden7A01.{1,2}.fq.gz`: individual 4 ddRAD seq. w/ umi barcodes

`testdata/ddRAD/popmap.txt`: population map used in Freebayes 

## Support

For further information or help, don't hesitate to get in touch on our [Slack `#radseq` channel](https://nfcore.slack.com/channels/radseq) (you can join with [this invite](https://nf-co.re/join/slack)).
